pyiron/atomistics/structure/atoms.py

```diff
@@ -3106,6 +3106,31 @@ def set_constraint(self, constrain):
             else:
                 self.selective_dynamics[atom_ind] = [False, False, False]
 
+    def apply_strain(self, epsilon, return_box=False):
+        """
+            Args:
+                epsilon (float/list/ndarray): epsilon matrix. If a single number is set, the same strain
+                                              is applied in each direction. If a 3-dim vector is set, it
+                                              will be multiplied by a unit matrix.
+                return_box (bool): whether to return a box. If set to True, only the returned box will 
+                                   have the desired strain and the original box will stay unchanged.
+        """
+        epsilon = np.array([epsilon]).flatten()
+        if len(epsilon) == 3 or len(epsilon) == 1:
+            epsilon = epsilon*np.eye(3)
+        epsilon.reshape(3,3)
+        if epsilon.min()<-1.0:
+            raise ValueError("Illegal strain value")
```

Comment:
nit: Can be more informative and let user know why it was illegal, i.e. too negative